jax.interpreters.pxla: remove deprecated functions:

jakevdp opened the pull request:
- `jax.interpreters.pxla.device_put`
- `jax.interpreters.pxla.make_sharded_device_array`

These were deprecated in JAX version 0.4.6, released 09 March 2023, greater than three months ago in accordance with the [API Compatibility policy](https://jax.readthedocs.io/en/latest/api_compatibility.html).

yashk2810 commented on CHANGELOG.md:
```diff
@@ -34,6 +34,11 @@ Remember to align the itemized text with the first line of an item within a list
     new `native_serializaation_disabled_checks` ({jax-issue}`#16347`).
+  * Two functions in `jax.interpreters.pxla` that have been deprecated since
+    JAX version 0.4.6 have now been removed:
+    * `device_put`: use `jax.device_put` instead.
```
Instead of just device_put, mention the full deprecated path. I know you have mentioned it above but while reading this I skipped that part and got confused 😛

jakevdp replied:
Sounds good – I structured it this way to avoid having to repeat `jax.interpreters.pxla`, but it probably makes sense just to write it directly.

jakevdp committed:
jax.interpreters.pxla: remove deprecated functions:

- jax.interpreters.pxla.device_put
- jax.interpreters.pxla.make_sharded_device_array
